src/symbol/mod.rs

```diff
@@ -1499,6 +1657,148 @@ impl<'t> TryFromCtx<'t, SymbolKind> for SeparatedCodeSymbol {
+    /// 
+    /// An array of variable size, currently only the first 4B are parsed.
+    pub rgl: u32,
+}
+
+impl<'t> TryFromCtx<'t, SymbolKind> for OemSymbol<'t> {
+    type Error = Error;
+
+    fn try_from_ctx(this: &'t [u8], _kind: SymbolKind) -> Result<(Self, usize)> {
+        let mut buf = ParseBuffer::from(this);
+
+        let symbol = OemSymbol {
+            id_oem: buf.parse_cstring()?,
+            type_index: buf.parse()?,
+            rgl: buf.parse()?,
+        };
+
+        Ok((symbol, buf.pos()))
+    }
+}
+
+/// Environment block split off from `S_COMPILE2`.
+///
+/// Symbol kind `S_ENVBLOCK`.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct EnvBlockSymbol<'t> {
+    /// EC flag (previously called `rev`).
+    pub edit_and_continue: bool,
+    /// Sequence of zero-terminated command strings.
+    pub rgsz: Vec<RawString<'t>>,
+}
+
+impl<'t> TryFromCtx<'t, SymbolKind> for EnvBlockSymbol <'t> {
+    type Error = Error;
+
+    fn try_from_ctx(this: &'t [u8], kind: SymbolKind) -> Result<(Self, usize)> {
+        let mut buf = ParseBuffer::from(this);
+        let flags: u8 = buf.parse()?;
+
+        let mut strings: Vec<RawString<'t>> = Vec::new();
+
+        while !buf.is_empty() {
+            strings.push(parse_symbol_name(&mut buf, kind)?);
+        }
+
+        let symbol = EnvBlockSymbol {
+            edit_and_continue: flags & 1 != 0,
+            rgsz: strings,
+        };
+
+        Ok((symbol, buf.pos()))
+    }
+}
+
+/// A COFF section in a PE executable.
+///
+/// Symbol kind `S_SECTION`.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct SectionSymbol<'t> {
+    /// Section number.
+    pub isec: u16,
+    ///  Alignment of this section (power of 2).
+    pub align: u8,
+    /// Reserved.  Must be zero.
+    pub reserved: u8,
+    /// Section's RVA.
+    pub rva: u32,
+    /// Section's CB.
+    pub cb: u32,
+    /// Section characteristics.
+    pub characteristics: u32,
+    /// Section name.
+    pub name: RawString<'t>
+
+}
+
+impl<'t> TryFromCtx<'t, SymbolKind> for SectionSymbol <'t> {
+    type Error = Error;
+
+    fn try_from_ctx(this: &'t [u8], kind: SymbolKind) -> Result<(Self, usize)> {
+        let mut buf = ParseBuffer::from(this);
+
+        let symbol = SectionSymbol {
+            isec: buf.parse()?,
+            align: buf.parse()?,
+            reserved: buf.parse()?,
+            rva: buf.parse()?,
+            cb: buf.parse()?,
+            characteristics: buf.parse()?,
+            name: parse_symbol_name(&mut buf, kind)?
+        };
+
+        Ok((symbol, buf.pos()))
+    }
+}
+
+/// A COFF section in a PE executable.
+///
+/// Symbol kind `S_SECTION`.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct CoffGroupSymbol<'t> {
+    /// COFF group's CB.
+    pub cb: u32,
+    /// COFF group characteristics.
+    pub characteristics: u32,
+    /// Symbol offset.
+    pub offset: PdbInternalSectionOffset,
+    /// Symbol segment.
+    pub segment: u16,    
```

Comment:
This part is wrong - The PdbInternalOffset is a combination of offset + segment, so the segment member should not appear here.